cosmo_tester/test_suites/cluster/full_cluster_test.py

```diff
@@ -34,3 +39,148 @@ def test_cluster_single_db(cluster_with_single_db, logger, attributes, cfy):
+
+    # the agent should now pick another broker
+    validate_cluster_and_agents(mgr1, expected_broker_status='Degraded')
+    new_agent_broker_ip = \
+        verify_agent_broker_connection_and_get_broker_ip(mgr1)
+    assert new_agent_broker_ip != agent_broker_ip
+
+    new_agent_broker = None
+    for broker in [broker1, broker2, broker3]:
+        if broker.private_ip_address == new_agent_broker_ip:
+            new_agent_broker = broker
+            break
+
+    # spin up 3 deployments
+    for i in range(3):
+        deployment_id = '{0}-{1}'.format(hello_world.deployment_id, i)
+        mgr1.client.deployments.create(
+            deployment_id=deployment_id,
+            blueprint_id=hello_world.blueprint_id,
+            inputs=hello_world.inputs)
+        mgr1.run_command(
+            'cfy executions start install -d {deployment_id} -t {tenant} '
+            '--timeout 900'.format(deployment_id=deployment_id,
+                                   tenant=hello_world.tenant))
+
+    # bring the stopped rabbit back up
+    with agent_broker.ssh() as broker_ssh:
+        broker_ssh.run('sudo service cloudify-rabbitmq start')
+
+    # verify that the resurrected rabbit shares the load
+    with new_agent_broker.ssh() as broker_ssh:
+        for broker in [broker1, broker2, broker3]:
+            num_queues = broker_ssh.run(
+                'sudo -u rabbitmq rabbitmqctl -n rabbit@{0} list_queues'
+                ' | wc -l'.format(broker.hostname))
+            assert num_queues > 0
+
+    # finally, uninstall all the hello worlds
+    mgr1.run_command(
+        'cfy executions start uninstall -d {deployment_id} -t {tenant} '
+        '--timeout 900'.format(deployment_id=hello_world.deployment_id,
+                               tenant=hello_world.tenant))
+    for i in range(3):
+        deployment_id = '{0}-{1}'.format(hello_world.deployment_id, i)
+        mgr1.run_command(
+            'cfy executions start uninstall -d {deployment_id} -t {tenant} '
+            '--timeout 900'.format(deployment_id=deployment_id,
+                                   tenant=hello_world.tenant))
+
+
+def prepare_cluster_with_agent(cluster_with_single_db, logger,
+                               module_tmpdir, attributes, ssh_key, cfy):
+    broker1, broker2, broker3, db, mgr1, mgr2 = cluster_with_single_db
+    configure_brokers_status_reporter([mgr1, mgr2],
+                                      [broker1, broker2, broker3])
+
+    logger.info('Installing a deployment with agents')
+    hello_world = centos_hello_world(cfy, mgr1, attributes, ssh_key,
+                                     logger, module_tmpdir)
+
+    _CloudifyManager.upload_necessary_files(mgr1)
+    _CloudifyManager.upload_necessary_files(mgr2)
+    _CloudifyManager.upload_plugin(mgr1,
+                                   mgr1._attributes.default_openstack_plugin)
+
+    # Install a hello world deployment.
+    # The test cluster has no outer network access so mgr1.use() won't do, so I
```

Comment:
mgr1.use() might work if you make the install use the floating IPs instead of the private ones?

Reply:
I didn't want to change the install since it's hard to predict how it would affect all the other system tests. Though we might have to change it anyway once we add a LB to the framework (my next task)

Reply:
Fair enough, I'll mark this as resolved.